widgets/grid/org.eclipse.nebula.widgets.grid/src/org/eclipse/nebula/widgets/grid/Grid.java

```diff
@@ -7776,9 +7784,11 @@ protected void setScrollValuesObsolete() {
 	int newColumn(final GridColumn column, final int index) {

```

Comment:
beside handling the `newColumn` you also have to handle the `removeColumn` and fix all indices